### ISSUE TITLE
Git stage / unstage (by line) no longer seem to update the file properly

### ISSUE THREAD

**Issue by eamodio** [Microsoft/vscode#26642]
- VSCode Version: Code - Insiders 1.13.0-insider (613724d43b4d50f6052bbb8007447ca96e687f26, 2017-05-16T06:11:37.391Z)
- OS Version: Windows_NT ia32 10.0.16193
---

Steps to Reproduce:

1. Open a file with changes
2. Attempt to stage a few lines
3. :bug: After the stage there isn't any indication of a change (like their used to be) where the stage lines are no longer "changes"
4. :bug: after you stage 1 line, it seems like any subsequent line stage/unstage operations have no effect

**Comment by joaomoreno** [comment]
Still works over here.

If you wait long enough on step 3 (10 seconds), do the changes come up?

**Comment by eamodio** [comment]
@joaomoreno anything else I can try on this -- still happens 100% of the time on any repo I've tried. I've also tried disabling all extensions and building vscode from source -- same behavior.

**Comment by eamodio** [comment]
@joaomoreno to add more details this works perfectly fine in `1.12` but still not in the latest nightly

**Comment by kumarharsh** [comment]
@joaomoreno - Git is indeed broken in recent flights.

Say I have a file with changes in lines: 1,2,3, 8,9,10

Selecting a few lines (1,2,3) in an updated file and staging them still highlights those lines. And even when I've committed those few lines, the Git view keeps highlighting the same lines (1,2,3, as well as 8,9,10) as having changes. But once I stage the file, the previously-committed lines (1,2,3) aren't highlighted anymore.

**Comment by kumarharsh** [comment]
![bug2](https://cloud.githubusercontent.com/assets/760112/26570360/c8008010-452d-11e7-8e1f-8ae088414ce0.gif)

**Comment by kumarharsh** [comment]
@joaomoreno I found a more sinister bug today: apparently if there are multiple hunks (say A,B,C,D), and I stage them one by one (A, then B, then C, then D), vscode **only** shows the last one (D) in the staged changes.

Now, If I **unselect the file**, and then reselect it, and stage other changes (A, then B, then C), again this same thing happens, but the original staged hunk is retained not replaced. So, now I have (A and B) in unstaged changes, and (C, and D) in staged changes.

So basically, for N hunks in my file which I want to stage, I have to **unselect my file N times** and stage one hunk each time :(

Seems a simple one to solve, but very unexpected - caused me to accidently "Clean changes" on a file I was sure I had staged all changes from :(

**Comment by joaomoreno** [comment]
Easily repros in my Windows box. Looking into it...

**Comment by opus131** [comment]
Why is this closed? And when was it fixed? I have v1.15.1 and I still have this issue.

**Comment by kumarharsh** [comment]
Yeah, it seems to have returned again @joaomoreno. The view in the `working tree` diff does not get updated when I stage a few lines.

**Comment by joaomoreno** [comment]
Can you reproduce in the latest [insider release](https://code.visualstudio.com/insiders)?

I can't, it seems to work for me.

**Comment by Nikki1993** [comment]
Having the same thing except VScode doesn't detect any changes to the files unless I do smth like git pull and then it wakes up. Running Archlinux.

**Comment by kumarharsh** [comment]
@joaomoreno yes I can reproduce it reliably. It's exactly as the GIF i posted above.

**Comment by joaomoreno** [comment]
Which OS, @kumarharsh?

@Nikki1993 #29821

**Comment by eamodio** [comment]
@joaomoreno I see it again on Windows in the latest insiders

**Comment by kumarharsh** [comment]
@joaomoreno Windows 10 Creators Update (same as before, I think you had reproduced it yourself too)

**Comment by joaomoreno** [comment]
Damn it, I think I did exactly the same mistake as before. Paths are hard, guys!

Created #34209, I'll make sure it goes into the stable recovery build.

**Comment by anteprimorac** [comment]
I still have this issue in 1.17.2 (macOS).
Git pull or any other command doesn't refresh gutter. Only reopening file refreshes gutter.

How I can see in git output it's refreshed after git show command had been called.

**Comment by anteprimorac** [comment]
Hi @joaomoreno ,
I recorded a screencast http://take.ms/Aa4xS. Let me know if you need more information.

**Comment by joaomoreno** [comment]
@anteprimorac I believe that is something else: https://github.com/Microsoft/vscode/issues/37417
